tock/api/views.py

```diff
@@ -358,3 +387,68 @@ def hours_by_quarter_by_user(request, *args, **kwargs):
         HoursByQuarterByUserSerializer(HoursByQuarterByUser(*each)).data
         for each in rows
     ])
+
+
+class TimecardPrefillDataSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = TimecardPrefillData
+        fields = (
+            'start_date',
+            'end_date',
+            'employee',
+            'employee_id',
+            'project',
+            'project_id',
+            'hours',
+            'notes',
+        )
+    project = serializers.CharField(source='project.name')
+    project_id = serializers.IntegerField(source='project.id')
+    employee = serializers.CharField(source='employee.user.username')
+    employee_id = serializers.IntegerField(source='employee.id')
+
+
+class TimecardsPrefillDataListView(generics.ListAPIView):
+    queryset = TimecardPrefillData.objects.all()
+    serializer_class = TimecardPrefillDataSerializer
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        serializer = self.get_serializer(queryset, many=True)
+
+        return Response(data={
+            "status": "200",
+            "data": serializer.data,
+        })
+
+
+class TimecardsPrefillDataUserListCreateView(generics.ListCreateAPIView):
+    serializer_class = TimecardPrefillDataSerializer
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        serializer = self.get_serializer(queryset, many=True)
+
+        return Response(data={
+            "status": "200",
+            "data": serializer.data,
+        })
+
+    def get_queryset(self):
+        if not self.request.user.is_superuser:
+            raise PermissionDenied
+
+        username = self.kwargs['username']
+
+        try:
+            userdata = UserData.objects\
+                       .prefetch_related('user')\
+                       .get(user__username=username)
+        except UserData.DoesNotExist:
+            raise Http404
+
+        return TimecardPrefillData.objects\
+               .prefetch_related('employee')\
+               .filter(employee_id=userdata.id)
```

Comment:
@rogeruiz, I think the source of the N+1 exceptions is here.  The queries can be tweaked a bit to pass the NPLUSONE checks:

```python
if not UserData.objects.filter(user__username=username).exists():
    raise Http404

return TimecardPrefillData.objects.filter(employee__user__username=username)\
.select_related('employee', 'project')
```